File: Baekjoon_Online_Judge/NO_02822/2822.py
# -*- encoding: utf-8 -*-
import sys
r_input = sys.stdin.readline

# ---------------------------- #
# | Created by happiness96   | #
# | Year 2020                | #
# | Month 08                 | #
# | Day 02                   | #
# | 2822 점수 계산             | #
# ---------------------------- #


if __name__ == "__main__":
    scores = []
    
    for ind in range(1, 9):
        score = int(r_input())

        scores.append([score, ind])
    
    scores.sort()
    total = 0
    res = []

    for ind in range(3, 8):
        total += scores[ind][0]
        res.append(scores[ind][1])
    
    print(total)
    print(*sorted(res))

File: Baekjoon_Online_Judge/NO_02822/2822.cpp
#include <iostream>
#include <algorithm>
#include <vector>
using namespace std;
/*
 ---------------------------- 
 | Created by happiness96   | 
 | Year 2020                | 
 | Month 08                 | 
 | Day 02                   | 
 | 2822 점수 계산             | 
 ---------------------------- 
*/

int main(){
    vector <pair <int, int>> scores;

    for(int i = 1; i < 9; i ++){
        int sc;

        cin >> sc;

        scores.push_back(make_pair(sc, i));
    }

    sort(scores.begin(), scores.end());

    int total = 0;
    int index[5];

    for(int i = 3; i < 8; i ++){
        total += scores[i].first;
        index[i - 3] = scores[i].second;
    }

    sort(index, index + 5);

    cout << total << '\n';
    for(auto ind : index) cout << ind << " ";

    return 0;
}
